### ISSUE TITLE
Render highway=pedestrian same as highway=living_street

### ISSUE THREAD

**Issue by jeisenbe** [gravitystorm/openstreetmap-carto#3961]
As mentioned in https://github.com/gravitystorm/openstreetmap-carto/issues/3849#issue-484899355 ("Render highway=pedestrian and highway=living_street same as highway=residential on z13") there is not a strong difference between the meaning of `highway=pedestrian` and `highway=living_street` - both are pedestrian-priority areas with limits on motor vehicle access and speed, but both almost always allow motor vehicle access at least at certain times of the day.

The current `highway=pedestrian` rendering is quite dark, with the same lightness as `landuse=residential`, `=retail`, `=commercial` and `=industrial`, so there is low contrast with the surrounding polygon fill in most places.

On the other hand, there are problems with the current `highway=living_street` fill, for example it is identical to the current `parking` area fill, which is a problem for areas. At z13 the color is hard to see against residential areas, according to https://github.com/gravitystorm/openstreetmap-carto/issues/3849#issue-484899355 

By combining the two features into one rendering, we would have one less different road color, which will make future changes and maintenance of the style easier. And we could adjust the color to compensate for the 3 problems mentioned above.

Rendering `highway=pedestrian` like `highway=living_street` would also mean showing these features at z13. Right now it is strange that cycleways are shown but not large pedestrian streets at z13.

Current rendering of Luxembourg city, z13 (no pedestrian roads, but living streets shown in southeast part of city centre):
![z13-luxembourg-current](https://user-images.githubusercontent.com/42757252/63674631-e8fecf00-c821-11e9-9a71-bff4c29eb9c2.png)

z14 Luxembourg current
![z14-luxembourg-city-current](https://user-images.githubusercontent.com/42757252/63674632-e9976580-c821-11e9-96f7-00bac7fc8d46.png)

**Comment by polarbearing** [comment]
> both almost always allow motor vehicle access at least at certain times of the day

I disagree. For hw=pedestrian that might be true if there is delivery access in a shopping zone, but there are other examples where they are indeed pedestrian only.

**Comment by ClarkstonCorrect** [comment]
I also thought that living street rendering doesn't stand out enough from landuse=residential.

**Comment by jeisenbe** [comment]
https://github.com/gravitystorm/openstreetmap-carto/pull/4555#issue-1245520424 - former maintainer @matkoniecz recommended to close this issue "as not worth doing" or perhaps "not improving things" - please explain?

**Comment by jeisenbe** [comment]
> this is not reasonable, a pedestrian highway is closed for regular (motor)vehicles, a living street is more or less a residential street with a particularly low speed limit
- @[dieterdreist](https://github.com/dieterdreist) in https://github.com/gravitystorm/openstreetmap-carto/issues/110#issuecomment-1150264386

If you are looking at a map, you would not expect to use a `highway=living_street` as a motor vehicle through route, right? But both would be good routes for pedestrians. Are they not practically similar to a general map user?

**Comment by Adamant36** [comment]
(I'm transferring this comment over from the other discussion)

> change the colour of living streets to something else (maybe a very light cyan, I don't think this will ever be confused with water given the context of these ways)

From what I've seen in America roads tagged as highway=living_street are just wrongly tagged residential roads. So if anything rendering for living streets should be removed because the tag clearly isn't used consistently or how it's supposed to be. If living streets even exist outside of a few very specific locations in the first place. I say that becaus I have yet to see the tag used on a street purely due to a sign saying it's a living street. Instead it always seems to come down to the opinion of an arm chair mapper based on satellite images.

**Comment by dieterdreist** [comment]
> If you are looking at a map, you would not expect to use a `highway=living_street` as a motor vehicle through route, right? But both would be good routes for pedestrians. Are they not practically similar to a general map user?

As a pedestrian they might be somewhat of similar utility (few nuisance from traffic) whilst still providing a different experience (in a pedestrian road you can (and everybody does) walk everywhere, there are no sidewalks and no parked cars. In a living street there are usually lots of parked cars and generally there will be some car traffic and people will tend to walk on the border, they are residential roads, in residential areas (pedestrian roads will be found in commercial/retail/central/scenic areas). While not being very convenient for longer distances, it may sometimes still be the best option to take a shortcut there. In my opinion, the current pedestrian symbology and the living street symbology should be inverted (living street lighter, pedestrian darker)

**Comment by pnorman** [comment]
> but both almost always allow motor vehicle access at least at certain times of the day.

This is not my experience with `highway=pedestrian`

I wouldn't be opposed to unifying the fill of the two because both existing fills have significant problems and I don't think it's possible to find two fills that will work well. At the same time, I don't think unifying the rendering is ideal.

> From what I've seen in America roads tagged as highway=living_street are just wrongly tagged residential roads

Living streets are, for a variety of reasons, a concept that has never been popular in the US.

**Comment by matkoniecz** [comment]
> I came to the impression that pedestrians are not in the focus group, in my city it

As person who designed this: it was intended to cover all modes of transport without blatant focus on any (I am rather on extreme side of opinion of role of a cars - and it is not one of praise).

Note that your screens are upscaled lower zoom levels, for view of that size of such area a different zoom level was tweaked to overall sort-of-work.

> >  but both almost always allow motor vehicle access at least at certain times of the day.

> This is not my experience with highway=pedestrian

+1

>  From what I've seen in America roads tagged as highway=living_street are just wrongly tagged residential roads

From what I know every single `highway=living_street` in USA is mistagged. `highway=living_street` requires pedestrians having priority over cars and being OK to walk on any part of the street.

**Comment by matkoniecz** [comment]
> former maintainer @matkoniecz recommended to close this issue "as not worth doing" or perhaps "not improving things" - please explain?

`highway=pedestrian` and `highway=living_street` substantially differ, and this provides useful feedback to mappers.

I would unify `highway=trunk` with `highway=primary` and `highway=tertiary` with `highway=secondary` before doing this.

**Comment by matkoniecz** [comment]
> these are cropped screenshots from the phone

and here are images without rescaling:

![screen08](https://user-images.githubusercontent.com/899988/172941298-62bf092a-1721-4f39-811d-390c2fc9ebdc.png)
![screen07](https://user-images.githubusercontent.com/899988/172941308-3d2522bb-1f91-4b8b-99c8-b5913842638a.png)

I consider them to work much better than what was presented

> and railways

I would need to check code to be sure, but rails/roads are not priviliged and ordered by layer order. Maybe except very low levels.

> it is done only with roads accessible to motor traffic 

I think that railways at least are really prominent compared to other maps, I would do the same with rivers but it would require preprocessing

**Comment by dieterdreist** [comment]
sent from a phone

> On 9 Jun 2022, at 22:49, Mateusz Konieczny ***@***.***> wrote:
> 
> I consider them to work much better than what was presented


different zoomlevel, your lower zoom is the same as my higher zoom (13), your higher zoom (14) is at the point where it starts working, it’s these lower zoomlevels (12/13) that suffer from the unfortunate decisions to omit pedestrian highways and buildings

**Comment by jeisenbe** [comment]
It looks like there is not much support for unifying the rendering of `highway=pedestrian` and `highway=living_street` at low zoom levels, but perhaps we can consider only at z13 and z14? I’ve changed the title of this issue.

**Comment by dieterdreist** [comment]
> It looks like there is not much support for unifying the rendering of `highway=pedestrian` and `highway=living_street` at low zoom levels, but perhaps we can consider only at z13 and z14? I’ve changed the title of this issue.

if you are going to conflate living street with another road class, my preference would be to render it the same as residential, this is what is similar

**Comment by bgo-eiu** [comment]
I am an American user of the pedestrian and living street tags and I only use pedestrian streets for streets which prohibit all vehicle traffic (this is how it is officially defined locally as well), and living street for streets which have some kind of physical barrier preventing general car traffic but may have residents park their cars there (for example, if there is a gate for cars but a public entrance for pedestrians, or some other sign that pedestrians are supposed to be able to walk in the street without mixing with general traffic). I think it's pointless to go back and forth about what the "correct" way to use the tags are when the understanding of the roads / concepts is different in different places, I am just saying I agree that living street is more similar to residential street just because it is usually possible for a vehicle to get on to them. There is no sign that will stop American drivers from driving on a street if it's wide enough for them, signs and road rules are just seen as suggestions here so that influences how people add features to the map.

**Comment by Adamant36** [comment]
> From what I know every single highway=living_street in USA is mistagged.

I was looking at the tags usage in South America and it doesn't seem much better there either. So I'd be interested to know where the tag is used properly outside of Europe. If the answer is that the tag is only used correctly in Europe then IMO it shouldn't be rendered for the same reasons that tags like [amenity=parcel_locker](https://github.com/gravitystorm/openstreetmap-carto/pull/4512) currently aren't, Or at least render it the same as residential roads. Otherwise, it just comes off like the style only caters to a small minority of mainly Western European users.
